### DISCUSSION TITLE
How I should contribute

### DISCUSSION THREAD

**Discussion by yyy33** [winnow-rs/winnow#589] [Ideas]
Hi, I've been reading and learning the winnow code recently and then had some thoughts. Should I just submit the pr or should I post issues for discussion first, I'm new to these processes, thanks!

**Comment by epage** [comment]
From https://github.com/winnow-rs/winnow/blob/main/CONTRIBUTING.md#pull-requests

> Looking for an idea? Check our [issues](https://github.com/winnow-rs/winnow/issues). If it's look more open ended, it is probably best to post on the issue how you are thinking of resolving the issue so you can get feedback early in the process. We want you to be successful and it can be discouraging to find out a lot of re-work is needed.
>
> Already have an idea? It might be good to first [create an issue](https://github.com/winnow-rs/winnow/issues/new) to propose it so we can make sure we are aligned and lower the risk of having to re-work some of it and the discouragement that goes along with that.

For background: as a maintainer, I find there is social pressure to move PRs forward, regardless of their state.  For my own health, I try to cut out that pressure by redirecting people away from them unless we are ready for it.  I've also found that multiple PRs may get opened and closed for the same thing over time and the conversation is much harder to follow.
